packages/ui-extensions/src/surfaces/point-of-sale/targets.ts

```diff
@@ -131,7 +136,7 @@ export interface ExtensionTargets {
     BlockComponents
   >;
   'pos.receipt-footer.block.render': RenderExtension<
-    StandardApi<'pos.receipt-footer.block.render'> & OrderApi,
+    StandardApi<'pos.receipt-footer.block.render'> & TransactionCompleteData,
```

Comment:
```suggestion
   TransactionCompleteData,
```

We need to remove the StandardApi because right now it's adding all of the extra sessionApi, toastApi, and a bunch of things that do "mutation" things to POS (that's why we consider that "API"). Even though this target renders and behaves a bit like a UI Extension Target, it's intended to be static, so we only need `TransactionCompleteData`, which extends `BaseData` and therefore will pass in connectivity, session, and other base objects

<img width="484" alt="Screenshot 2025-03-13 at 1 26 22 PM" src="https://github.com/user-attachments/assets/a12fa20b-83b3-49ac-86f3-d83a46bb8179" />

Reply:
Oh, and also, it is `TransactionCompleteData | {transaction: ReprintReceiptData}`

Reply:
ok, removed!
that's interesting now my autocomplete shows `transaction` too. wasn't able to get it yesterday 🤔

Reply:
You'll also need {key: string, value: striing} to make TS happy
```js
'pos.customer-details.block.render': RenderExtension<
    {
      [key: string]: any;
    } & (TransactionCompleteData | {transaction: ReprintReceiptData}),
    BlockComponents
  >;
```

Reply:
good call on the `ReprintReceiptData`, but trying to avoid the `any` object since that is a hack for the current `useApi` usage.

looks like there's issues with the typing after all. will try to find solution for abit if not just mass allow like that.